plugins/linux-pipewire/pipewire.c

```diff
@@ -46,6 +46,26 @@ struct obs_pw_version {
 	int micro;
 };
 
+typedef struct _obs_pipewire_data obs_pipewire_data;
+
+// This struct contains the shared fields between all obs_pipewire wrappers
+// The C Standard guarantees that the memory layout off these parts of the
+// struct are identical and as such we can cast any extended struct to this
+// and use the fields without issues. Any change here has to be added to the
+// extended structs and the order of fields has to be preserved.
```

Comment:
I don't think this is a good approach for sharing struct fields. Traditionally, if you want `obs_pipewire_stream_data` to extend `obs_pipewire_data`, you'd do something like:

```c
struct _obs_pipewire_data {
	... fields ...
};

struct _obs_pipewire_stream_data {
	obs_pipewire_data base;

	... other fields ...
};
```

Then, in the places where you want to treat `obs_pipewire_stream_data`  as a `obs_pipewire_data`, you'd do:

```c
obs_pipewire_data *data = &stream_data->base;
```